mgmt_cli_test.py

```diff
@@ -461,6 +464,45 @@ def test_repair_multiple_keyspace_types(self):  # pylint: disable=invalid-name
         self.log.info("the sctool repair command was completed successfully")
         self.log.info('finishing test_repair_multiple_keyspace_types')
 
+    def test_enospc_during_backup(self):
+        self.log.info('starting test_enospc_during_backup')
+        if not self.is_cred_file_configured:
+            self.update_config_file()
+        manager_tool = mgmt.get_scylla_manager_tool(manager_node=self.monitors.nodes[0])
+        hosts = self.get_cluster_hosts_ip()
+        location_list = [f's3:{self.bucket_name}']
+        selected_host = hosts[0]
+        mgr_cluster = manager_tool.get_cluster(cluster_name=self.CLUSTER_NAME) \
+            or manager_tool.add_cluster(name=self.CLUSTER_NAME, host=selected_host,
+                                        auth_token=self.monitors.mgmt_auth_token)
+
+        target_node = self.db_cluster.nodes[1]
+
+        self.generate_load_and_wait_for_results()
+        has_enospc_been_reached = False
+        with DbEventsFilter(type='NO_SPACE_ERROR', node=target_node),\
+                DbEventsFilter(type='BACKTRACE', line='No space left on device', node=target_node), \
+                DbEventsFilter(type='DATABASE_ERROR', line='No space left on device', node=target_node), \
+                DbEventsFilter(type='FILESYSTEM_ERROR', line='No space left on device', node=target_node):
+            try:
+                backup_task = mgr_cluster.create_backup_task(location_list=location_list)
+                backup_task.wait_for_status(list_status=[TaskStatus.RUNNING], step=60)
+                backup_task.stop()
+
+                has_enospc_been_reached = True
+                Nemesis.reach_enospc_on_node(target_node=target_node, log_object=self.log)
+
+                backup_task.start(cmd=f"task -c {backup_task.cluster_id} start {backup_task.id} --continue=true")
+
+                backup_task.wait_and_get_final_status()
+                assert backup_task.status == TaskStatus.DONE, "The backup failed to run on a node with no free space," \
```

Comment:
it will only succeed if the snapshots were already taken, otherwise, there will be no space left on disk to write the snapshots